frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -9,36 +10,34 @@ import {
   useState,
 } from 'react';
 
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
 import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
 import { StakingProgramId } from '@/enums/StakingProgram';
-import { useAgent } from '@/hooks/useAgent';
-import { useChainId } from '@/hooks/useChainId';
-import { useStakingContractDetailsByStakingProgram } from '@/hooks/useStakingContractDetails';
+import { useServiceId } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { useStakingPrograms } from '@/hooks/useStakingPrograms';
 import { StakingContractDetails } from '@/types/Autonolas';
 
-import {
-  INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  StakingProgramContext,
-} from './StakingProgramProvider';
+import { StakingProgramsContext } from './StakingProgramsProvider';
 
 /**
  * hook to get all staking contract details
  */
 const useAllStakingContractDetails = () => {
-  const stakingPrograms = [INITIAL_DEFAULT_STAKING_PROGRAM_ID];
-  const chainId = useChainId();
-  const currentAgent = useAgent();
+  const { allStakingProgramIds } = useStakingPrograms();
+  const { selectedAgentConfig } = useServices();
+  const { serviceApi, homeChainId } = selectedAgentConfig;
```

Comment:
why calling it homeChainId?

Reply:
decision made by middleware team, home chain is where the NFT is staked
Predict Trader is Gnosis
Optimus is Optimism (but also has safes and eoas on other chains)